tests/test_management_commands.py

```diff
@@ -0,0 +1,481 @@
+
+    def test_minutes(self):
+        expected_datemodified = datetime.datetime.utcnow() - datetime.timedelta(
+            minutes=30
+        )
+        output = self.call_command(minutes=30)
+
+        # Check the command output
+        self.assertIn(
+            "Looking for document assets modified in the last 30 minute(s)", output
+        )
+
+        # Check the mocked API was called as expected
+        self.mock_api_client.asset_bank_client.media_list.assert_called_once_with(
+            {
+                "dateModified": expected_datemodified.strftime("%Y-%m-%dT%H:%M:%SZ"),
+                **self.common_api_kwargs,
+            }
+        )
+
+        # Check the patched object was updated and saved as expected
+        self.patched_obj.update_from_asset_data.assert_called_once_with(TEST_ASSET_DATA)
+        self.patched_obj.save.assert_called_once()
+
+    def test_hours(self):
+        expected_datemodified = datetime.datetime.utcnow() - datetime.timedelta(hours=3)
+        output = self.call_command(hours=3)
+
+        # Check the command output
+        self.assertIn(
+            "Looking for document assets modified in the last 3 hour(s)", output
+        )
+
+        # Check the mocked API was called as expected
+        self.mock_api_client.asset_bank_client.media_list.assert_called_once_with(
+            {
+                "dateModified": expected_datemodified.strftime("%Y-%m-%dT%H:%M:%SZ"),
+                **self.common_api_kwargs,
+            }
+        )
+
+        # Check the patched object was updated and saved as expected
+        self.patched_obj.update_from_asset_data.assert_called_once_with(TEST_ASSET_DATA)
+        self.patched_obj.save.assert_called_once()
+
+    def test_days(self):
+        expected_datemodified = datetime.datetime.utcnow() - datetime.timedelta(days=3)
+        output = self.call_command(days=3)
+
+        # Check the command output
+        self.assertIn(
+            "Looking for document assets modified in the last 3 day(s)", output
+        )
+
+        # Check the mocked API was called as expected
+        self.mock_api_client.asset_bank_client.media_list.assert_called_once_with(
+            {
+                "dateModified": expected_datemodified.strftime("%Y-%m-%dT%H:%M:%SZ"),
+                **self.common_api_kwargs,
+            }
+        )
+
+        # Check the patched object was updated and saved as expected
+        self.patched_obj.update_from_asset_data.assert_called_once_with(TEST_ASSET_DATA)
+        self.patched_obj.save.assert_called_once()
+
+
+@freeze_time()
+class UpdateStaleVideosTestCase(BaseSyncCommandTestCase):
+    """
+    Unit tests for the 'update_stale_videos' management command.
+    """
+
+    command_name = "update_stale_videos"
+    model_class = get_video_model()
+    common_api_kwargs = {
+        "orderBy": "dateModified desc",
+        "page": 1,
+        "limit": 200,
+        "type": "video",
+    }
```

Comment:
the doc and video testcases are pretty basically the images test case with 2 differences:

1. the message look for "documents" and "videos" correspondingly, rather than "images", so easy to parametrize
2. both are missing
```
self.mock_api_client.asset_bank_client.media_info.assert_called_once_with(TEST_ASSET_ID)
```

in a few places because the images command has https://github.com/torchbox/wagtail-bynder/blob/main/src/wagtail_bynder/management/commands/update_stale_images.py#L14

I wonder if we can parametrize this in `BaseSyncCommandTestCase` so the specific test
case just have `command_name`, `model_class`, `common_api_kwargs`, the message and whether `media_info` is called.

DRY and all that

Reply:
Yes, I agree. I considered writing a custom assertion quite early on, but felt like it was too early at that point. Now I know the shape of everything, it's worth revisiting.